AUTH.md

```diff
@@ -0,0 +1,304 @@
+  requests against any garden or system.
+
+## Authentication Basics
+
+When authorization is enabled on your garden, users need a way to authenticate
+in order to access the garden using their account. From a user's perspective,
+this is done via the following:
+
+### Web UI Access
+
+When authorization is enabled, a "login" button will appear toward the top right
+of the page. Clicking this brings up a login box for the user to enter their
+credentials and sign in.
+
+### API Access
+
+For direct API access, a user would first send their credentials in a POST to
+`/api/v1/token`. This results in an access and refresh token being provided back
+in the response. The access token would then be provided by the user in
+subsequent API calls via the `Authorization: Bearer` header.
+
+This token retrieval and usage is handled for the user by the Web UI and
+EasyClient, but both use this API login and access token workflow behind the
+scenes.
+
+## Auth Settings
+
+Authorization and authentication settings are housed under a top level `auth`
+section of the main application configuration yaml file. The available settings
+are:
+
+### authentication_handlers
+
+This section allows you to configure the ways that users are able to
+authenticate to the garden. The available handlers and their respective settings
+are:
+
+#### basic
+
+Allows for username and password authentication. Available settings:
+
+- **enabled:** Set to `true` to allow username and password authentication.
+
+#### trusted_header
+
+Allow for users to be authenticated via request headers that get set via a
+trusted proxy. If your garden sits behind a proxy that will authenticate the
+user and place their username in a header, enabling this will tell the garden to
+use that provided username.
+
+When logging in via the UI, the login dialog will always show the input fields
+for username and password. If a user is authenticating by providing certificates
+on all requests that go through a proxy, these fields can safely be left blank.
+Since the trusted headers will be included on the login request, the user will
+still be able to login.
+
+**NOTE:** If you enable this it is imperitive that users are required to access
+your garden through the proxy and do not have a means of accessing the garden
+directly. Direct garden access could allow users to set what are supposed to be
+trusted headers themselves, allowing them to masquerade as whomever they wish.
+
+Available settings:
+
+- **create_users:** When `true` an account will be created automatically for a
+  user upon first access. When `false` the user will be denied access unless an
+  account already exists for them.
+- **enabled:** Set to `true` to allow trusted header authentication.
+- **user_groups_header:** The name of the header that will contain a comma
+  separated list of the user's group memberships.
+- **username_header:** The name of the header that will contain the username.
```

Comment:
Do these header settings have default values if the keys are absent or left blank?

Reply:
It turns out these did have a default.  I added info.